cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/ConfigManager.kt

```diff
@@ -10,15 +10,20 @@ import io.vertx.core.json.JsonArray
 import io.vertx.core.json.JsonObject
 import io.vertx.core.logging.Logger
 import io.vertx.core.logging.LoggerFactory
+import java.util.*
 
 class ConfigManager : AbstractVerticle() {
 
     private val widgets = mutableMapOf<String, Widget>()
     private lateinit var storage: Storage
     private lateinit var endpoints: JsonArray
+    private lateinit var credentials: JsonArray
+    private lateinit var endpoint: Endpoint
 
     override fun start() {
-        endpoints = config().getJsonArray("endpoints")
+        endpoints = config().getJsonArray(ENDPOINTS)
```

Comment:
this is know confusing - in my opinion there is no need to store `endpoints`, `credentials` and most importantly `endpoint` here. Look below to see how I see it: